contracts/base/FallbackManager.sol

```diff
@@ -0,0 +1,57 @@
+pragma solidity ^0.5.0;
+
+import "../common/SelfAuthorized.sol";
+
+/// @title Fallback Manager - A contract that manages fallback calls made to this contract
+/// @author Richard Meissner - <richard@gnosis.pm>
+contract FallbackManager is SelfAuthorized {
+
+    // TO BE DISCUSSED: Using this with assembly storage access costs about 100 gas more for a read, and 50 for a write,
+    // but is future proof for variable order changes in the future
+    // TODO: set hash to keccak("gnosis_safe_fallback_handler")
+    bytes32 internal constant FALLBACK_HANDLER_STORAGE_SLOT = 0x035aff83d86937d35b32e04f0ddc6ff469290eef2f1b692d8a815c89404d4749;
+
+    function internalSetFallbackHandler(address handler) internal {
+        bytes32 slot = FALLBACK_HANDLER_STORAGE_SLOT;
+        // solium-disable-next-line security/no-inline-assembly
+        assembly {
+            sstore(slot, handler)
+        }
+    }
+
+    /// @dev Allows to add a contract to handle fallback calls.
+    ///      Only fallback calls without value and with data will be forwarded.
+    ///      This can only be done via a Safe transaction.
+    /// @param handler contract to handle fallbacks calls.
+    function setFallbackHandler(address handler)
+        public
+        authorized
+    {
+        internalSetFallbackHandler(handler);
+    }
+
+    function ()
+        external
+        payable
+    {
+        // Only calls without value and with data will be forwarded
+        if (msg.value > 0 || msg.data.length == 0) return;
```

Comment:
Should we also have a handler for ether transfers? ... only real purpose would to to reject ether transfers ... or emit events

Reply:
For me the default version shouldn't have it (it will require even more gas to send ether to the Safe, as the proxy and this function will be called) and I just see it useful for some cases that can implement their own module.

Reply:
Yeah, many clients like metamask already have issues sending ETH to a safe because hardcode the gas sent to 21k instead of estimating it or adding a bit more.

I think is that edge case that would be desirable that it can be left out

Reply:
I don't see a need for adding an event for now and we want to allow Ether transfers per default.